src/rust/src/padding.rs

```diff
@@ -110,6 +109,62 @@ impl PKCS7PaddingContext {
     }
 }
 
+#[pyo3::pyclass]
+pub(crate) struct ANSIX923PaddingContext {
+    block_size: usize,
+    buffer: Option<Vec<u8>>,
+}
+
+#[pyo3::pymethods]
+impl ANSIX923PaddingContext {
+    #[new]
+    pub(crate) fn new(block_size: usize) -> ANSIX923PaddingContext {
+        ANSIX923PaddingContext {
+            block_size: block_size / 8,
+            buffer: Some(Vec::new()),
+        }
+    }
+
+    pub(crate) fn update<'p>(
+        &mut self,
+        buf: CffiBuf<'p>,
+        py: pyo3::Python<'p>,
+    ) -> CryptographyResult<pyo3::Bound<'p, pyo3::types::PyBytes>> {
+        match self.buffer.as_mut() {
+            Some(v) => {
+                v.extend_from_slice(buf.as_bytes());
+                let finished_blocks = v.len() / (self.block_size);
+                let result_size = finished_blocks * (self.block_size);
+                let result = v.drain(..result_size);
+                Ok(pyo3::types::PyBytes::new(py, result.as_slice()))
+            }
+            None => Err(exceptions::already_finalized_error()),
+        }
+    }
+
+    pub(crate) fn finalize<'p>(
+        &mut self,
+        py: pyo3::Python<'p>,
+    ) -> CryptographyResult<pyo3::Bound<'p, pyo3::types::PyBytes>> {
+        match self.buffer.take().as_mut() {
+            Some(v) => {
+                let pad_size = (self.block_size).saturating_sub(v.len());
+                let mut count = 0;
+                v.extend(std::iter::from_fn(move || {
+                    count += 1;
+                    match count.cmp(&pad_size) {
+                        std::cmp::Ordering::Less => Some(0),
+                        std::cmp::Ordering::Equal => Some(pad_size as u8),
+                        std::cmp::Ordering::Greater => None,
+                    }
+                }));
```

Comment:
I think  `for _ in 0..(pad_size - 1) { v.push(0) }; v.push(pad_size);` would be simpler and easier to read.

Reply:
Did a slight variation on this: create a zero-filled vector and then do `v.push(pad_size)`